python/cudf/cudf/options.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from collections.abc import Container
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional
+
+
+@dataclass
+class Option:
+    default: Any
+    value: Any
+    description: str
+    validator: Callable
+
+
+_OPTIONS: Dict[str, Option] = {}
+
+
+def _register_option(
+    name: str, default_value: Any, description: str, validator: Callable
+):
+    """Register an option.
+
+    Parameters
+    ----------
+    name : str
+        The name of the option.
+    default_value : Any
+        The default value of the option.
+    description : str
+        A text description of the option.
+    validator : Callable
+        Called on the option value to check its validity. Should raise an
+        error if the value is invalid.
+    """
+    validator(default_value)
+    _OPTIONS[name] = Option(
+        default_value, default_value, description, validator
+    )
+
+
+def get_option(name: str) -> Any:
+    """Get the value of option.
+
+    Parameters
+    ----------
+    key : str
+        The name of the option.
+
+    Returns
+    -------
+    The value of the option.
+    """
+    return _OPTIONS[name].value
+
+
+def set_option(name: str, val: Any):
+    """Set the value of option.
+
+    Raises ``ValueError`` if the provided value is invalid.
```

Comment:
This should go into a `Raises` section. Additionally, we should document the error if `name` is not a valid option.